Providers/Profile.swift

```diff
@@ -606,9 +638,11 @@ public class BrowserProfile: Profile {
         var isSyncing: Bool {
             syncLock.lock()
             defer { syncLock.unlock() }
-            return !(syncReducer?.isFilled ?? true)
+            return syncDisplayState != nil && syncDisplayState! == .InProgress
```

Comment:
I'm really nervous `isSyncing` using a display state to tell if we're syncing internally (at L1071 especially). e.g. `onStartSyncing` seems to affect `isSyncing` but doesn't do anything outside of change status, and isn't protected by the lock this is.